Examples/MagicWeatherSwiftUI/Shared/Sources/ViewModels/UserViewModel.swift

```diff
@@ -26,6 +26,23 @@ class UserViewModel: ObservableObject {
     /* Set from the didSet method of customerInfo above, based on the entitlement set in Constants.swift */
     @Published var subscriptionActive: Bool = false
     
+    /* Keep track of the task that listens for changes to the `customerInfoStream` so that it can be cancelled later on */
+    private var task = Task<Void, Never> {}
+    
+    private init() {
+        /* Listen to changes in the `customerInfo` object using an `AsyncStream` */
+        self.task = Task.detached {
```

Comment:
Does it really need to be a `detached` `Task`? 

If I'm not mistaken, if it weren't a `detached` task, we wouldn't have to keep the reference to `task` and we wouldn't need to `cancel()` it on `deinit`. And it would still be canceled when `UserViewModel` got deallocated.

Not very important, but it would simplify the code a bit.

Reply:
I don't think it does tbf, let's go for simplicity for this. Let me update the code now 👀